src/components/Utils/helpers.js

```diff
@@ -37,3 +37,52 @@ export const latestInterval = (
 export const isSubstring = (fullText = '', textToFind = '') => {
   return fullText.toLowerCase().includes(textToFind.toLowerCase());
 };
+
+export function differenceInDays(dateOne, dateTwo) {
+  const difference = dateOne.getTime() - dateTwo.getTime();
+  const days = Math.ceil(difference / (1000 * 3600 * 24));
+  return days;
+}
+
+export const getDaysNextPurchase = (product) => {
+  const havePurchases = product.numberOfPurchases > 0;
+
+  const estimatedDays = havePurchases
+    ? product.estimatedDaysNextPurchase
+    : product.nextPurchase;
+  const lastPurchase = havePurchases
+    ? product.lastPurchasedDate.toDate()
+    : product.creationDate.toDate();
+
+  const dateNextPurchase = new Date(lastPurchase);
+  dateNextPurchase.setDate(lastPurchase.getDate() + estimatedDays);
+
+  const today = new Date();
+  const daysNextPurchase = differenceInDays(dateNextPurchase, today);
+
+  return daysNextPurchase;
+};
+
+export const sortProducts = (p1, p2) => {
+  let order = p1.daysNextPurchase - p2.daysNextPurchase;
+  if (order === 0) order = p1.item > p2.item ? 1 : -1;
```

Comment:
It's very interesting, although I'm curious, will there is some problem if you extract this logic to two different functions such as sortProductsByDaysNextPurchase and sortProductsByName. Because I believe this function sorts two times 😄 🤔 .

Reply:
Exactly Alejandro, because it should not only be ordered alphabetically but also by number of days of next purchase. I remind you that there are different ways to solve a problem. Why do two functions if you can do one.

Reply:
FYI, this function is not sorting two times, it sorts once, based on two criteria :)